mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -449,6 +449,16 @@ class SparseVector(
   override def toString: String =
     "(%s,%s,%s)".format(size, indices.mkString("[", ",", "]"), values.mkString("[", ",", "]"))
 
```

Comment:
Thanks @srowen for the comment. Glad to discuss it with someone.

_Vector: override def hashCode(): Int = util.Arrays.hashCode(this.toArray)_
 
I understand it's the general guideline to override `hashCode` at the same time.
Yet intentionally or not, the original code promises that `DenseVector` and `SparseVector` would return the same results of `equals` and `hashCode` for the same array content. And that makes some senses.

As in the description of the PR, I don’t want to introduce breaking changes. And if we want to keep the original design, the current implementation of `hashCode` in `Vector` is one of the best choices. That’s why `hashCode` was intentionally left out of the PR. (maybe I should add some comment)